test/tools/test_tools_memories.py

```diff
@@ -58,3 +62,415 @@ def test_sopel_identifier_memory_channel_str():
+    assert len(memory) == 2
+
+    # verify key-exists behavior w/explicit default of None
+    assert memory.get('somecamelcase', None) is True
+    assert memory.get('LowerOnly', None) is False
+    assert len(memory) == 2
+
+    # verify key-exists behavior w/explicit "real" default value
+    assert memory.get('somecamelcase', 'DEFAULT') is True
+    assert memory.get('LowerOnly', 'DEFAULT') is False
+    assert len(memory) == 2
+
+    # verify key-missing behavior w/implicit default
+    assert 'missing_key' not in memory
+    assert memory.get('missing_key') is None
+    assert len(memory) == 2
+
+    # verify key-missing behavior w/explicit default of None
+    assert 'missing_key' not in memory
+    assert memory.get('missing_key', None) is None
+    assert len(memory) == 2
+
+    # verify key-missing behavior w/explicit "real" default value
+    assert 'missing_key' not in memory
+    assert memory.get('missing_key', 'DEFAULT') == 'DEFAULT'
+    assert len(memory) == 2
+
+
+def test_sopel_identifier_memory_pop():
+    memory = memories.SopelIdentifierMemory()
+    memory['SomeCamelCase'] = True
+    assert len(memory) == 1
+
+    # verify SopelIdentifierMemory.pop('key') behavior
+    pop_me = memory.copy()
+    assert len(pop_me) == 1
+    assert pop_me.pop('someCAMELcase') is True
+    assert len(memory) == 1  # sanity check
+    assert len(pop_me) == 0
+
+    # verify SopelIdentifierMemory.pop('key', None) behavior
+    pop_me = memory.copy()
+    assert len(pop_me) == 1
+    assert pop_me.pop('someCAMELcase', None) is True
+    assert len(pop_me) == 0
+
+    # verify SopelIdentifierMemory.pop('key', 'default_value') behavior
+    pop_me = memory.copy()
+    assert len(pop_me) == 1
+    assert pop_me.pop('someCAMELcase', 'DEFAULT') is True
+    assert len(pop_me) == 0
+
+    # verify SopelIdentifierMemory.pop('missing_key') behavior
+    pop_me = memory.copy()
+    assert len(pop_me) == 1
+    assert 'missing_key' not in pop_me
+    with pytest.raises(KeyError):
+        pop_me.pop('missing_key')
+    assert len(pop_me) == 1
+
+    # verify SopelIdentifierMemory.pop('missing_key', None) behavior
+    assert 'missing_key' not in pop_me
+    assert pop_me.pop('missing_key', None) is None
+    assert len(pop_me) == 1
+
+    # verify SopelIdentifierMemory.pop('missing_key', 'default_value') behavior
+    assert 'missing_key' not in pop_me
+    assert pop_me.pop('missing_key', 'DEFAULT') == 'DEFAULT'
+    assert len(pop_me) == 1
+
+
+def test_sopel_identifier_memory_del_key():
+    memory = memories.SopelIdentifierMemory()
+    memory['KeY'] = True
+    assert 'KeY' in memory
+    del memory['KeY']
+    assert 'KeY' not in memory
+
+
+def test_sopel_identifier_memory_from_dict():
+    dictionary = {'CamelCaseKey': True, 'lowercasekey': False}
+    assert 'CamelCaseKey' in dictionary
+    assert 'camelcasekey' not in dictionary
+    assert 'lowercasekey' in dictionary
+    assert 'LowerCaseKey' not in dictionary
```

Comment:
Don't need those here, that's how `dict`  are supposed to work already. Trust Python!

```suggestion
    dictionary = {'CamelCaseKey': True, 'lowercasekey': False}
```